packages/flutter/lib/src/rendering/debug_overflow_indicator.dart

```diff
@@ -6,6 +6,8 @@ import 'dart:math' as math;
 import 'dart:ui' as ui;
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/src/widgets/framework.dart';
```

Comment:
This is a layering violation - rendering shouldn't know about widgets. I would ask the framework folks what a better way to structure this would be.

Reply:
@goderbauer do you have some guidance here? we need access to some of the code in `widget_inspector.dart`

Reply:
Could you utilise a DiagnosticPropertiesTransformer (added to FlutterErrorDetails.propertiesTransformers [1]) to change the information collected in a lower level to something more actionable in a higher level?

We use this mechanism to enrich errors in the rendering layer with additional information from the widget layer.

[1] https://master-api.flutter.dev/flutter/foundation/FlutterErrorDetails/propertiesTransformers.html

Reply:
It looks like this method already dips into that mechanism: line  265 below adds a DiagnosticsDebugCreator, which gets processed by the widget_insepctor (via one of those transformers) here: https://github.com/flutter/flutter/blob/3fd3447f9676b71d91ee81e32f335c702cef8e7c/packages/flutter/lib/src/widgets/widget_inspector.dart#L2840

You may be able to reuse that transformer (or add a new one for your purposes).